strictlyvc_news_collector.py

```
@@ -268,36 +269,58 @@ def extract_from_parts(parts):
-            text_lower = text.lower()
+        # Import section headers from news_keywords
+        from news_keywords import SECTION_HEADERS
+        section_patterns = [header.lower() for header in SECTION_HEADERS]
+
+        # Get all text content from the email
+        all_text = soup.get_text()
+
+        # Find sections by looking for section headers
+        for pattern in section_patterns:
+            # Search for the pattern in the HTML - look for text containing the pattern
+            for element in soup.find_all(string=lambda text: text and text.strip().lower() in [pattern, pattern + ':', pattern + ' '] if text else False):
```

Ruff (0.12.2)

Function definition does not bind loop variable `pattern`

(B023)